pkg/mimirtool/analyze/grafana.go

```diff
@@ -140,10 +158,21 @@ func metricsFromTemplating(templating minisdk.Templating, metrics map[string]str
 
 // Workaround to support Grafana "timeseries" panel. This should
 // be implemented in grafana/tools-sdk, and removed from here.
-func getCustomPanelTargets(panel minisdk.Panel) *[]minisdk.Target {
+func getCustomPanelTargets(panel minisdk.Panel, datasourceUID string) *[]minisdk.Target {
 	if panel.CommonPanel.Type != "timeseries" {
 		return nil
 	}
+	if datasourceUID != "" {
+		if panel.Datasource != nil {
+			// we'll filter mixed targets later
+			if panel.Datasource.Type != "datasource" && panel.Datasource.UID != datasourceUID {
```

Comment:
> The docs for this say that the type should be a value like `prometheus` or `graphite`. Are the docs wrong?

You want to use `== "prometheus"` instead of `!="datasource"` ? Because we're filtering non-mixed targets here, mixed will be filtered later. Mixed looks like this - https://github.com/grafana/mimir/pull/4790/files#diff-068fcc61ddfdbbc3578dda5045a468fb0488e436ee77057cbfc187fa0da4f8f8R110-R112

Reply:
> I don't think it will be much harder to also support the legacy format. Maybe we can add a method on DatasourceRef - DatasourceID() string, which returns the right identificator - name or UID

Makes sense - but I'm not sure how mixed datasources looks like in legacy format. Let me check that first.

Reply:
I'm sorry I had misread what the `if` does. `panel.Datasource.Type != "datasource"` is correct 👍 

But this also made me think about non-prometheus dashboards. What if there is a loki panel with its own loki datasource - will `mimirtool analyze dashboard` also extract its query?

Reply:
It should. Type == datasource is present in all mixed datasources, rest is done by parser.